lib/src/model/leg.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:sublin_tomp_model/src/model/asset.dart';
+import 'package:sublin_tomp_model/src/model/asset_type.dart';
+import 'package:sublin_tomp_model/src/model/condition.dart';
+import 'package:sublin_tomp_model/src/model/fare.dart';
+import 'package:sublin_tomp_model/src/model/leg_state.dart';
+import 'package:sublin_tomp_model/src/model/place.dart';
+import 'package:sublin_tomp_model/src/model/suboperator.dart';
+import 'package:sublin_tomp_model/src/model/token.dart';
+
+part 'leg.g.dart';
+
+@JsonSerializable()
+class Leg {
+  final String? id;
+  final Place from;
+  final Place? to;
+  final DateTime? departureTime;
+  final DateTime? arrivalTime;
+  final List<String>? travelerReferenceNumbers;
+  final AssetType? assetType;
+  final int? legSequenceNumber;
+  final Asset? asset;
+  final Fare? pricing;
+  final Suboperator? suboperator;
+  final List<Condition>? conditions;
+  @LegStateSerialiser()
+  final LegState? state;
+  final Duration? departureDelay;
+  final Duration? arrivalDelay;
+  final int? distance;
+  final List<List<double>>? progressGeometry;
+  final Token? ticket;
+  final List<Token>? assetAccessData;
```

Comment:
Asset Access data should be a list of many or on element, and not one element or a list